core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -524,6 +528,14 @@ class SparkContext(config: SparkConf) extends Logging {
     executorEnvs ++= _conf.getExecutorEnv
     executorEnvs("SPARK_USER") = sparkUser
 
+    val configuredPluginClasses = conf.get(SHUFFLE_IO_PLUGIN_CLASS)
+    val maybeIO = Utils.loadExtensions(
+      classOf[ShuffleDataIO], Seq(configuredPluginClasses), conf)
+    require(maybeIO.size == 1, s"Failed to load plugins of type $configuredPluginClasses")
```

Comment:
Let's try to make this more descriptive - e.g. report the classes that were bad if there were multiple, or mention that no class was found if there are none.

Reply:
I think if they were bad, then `Utils.loadExtensions()` will throw, so that case is covered. I can add another check before this to check if there were 0 classes found, in which case, it'll log a different message